PowerEditor/src/Notepad_plus.cpp

```diff
@@ -3103,14 +3103,25 @@ void Notepad_plus::maintainIndentation(TCHAR ch)
 	intptr_t tabWidth = _pEditView->execute(SCI_GETTABWIDTH);
 
 	LangType type = _pEditView->getCurrentBuffer()->getLangType();
+	LangAutoIndentType autoIndentType = LangAutoIndentType::Standard;
+
+	// For external languages, query for custom auto-indentation funcionality
+	if (type >= L_EXTERNAL)
+	{
+		NppParameters& nppParam = NppParameters::getInstance();
+		autoIndentType = nppParam.getELCFromIndex(type - L_EXTERNAL)._autoIndentType;
+		if (autoIndentType == LangAutoIndentType::Custom)
+			return;

```

Comment:
Why return for `Custom` mode?
Does `Custom` mode is for telling Notepad++ "do nothing about indentation, I (external lexer) will do it" ?

Reply:
Yes, that's it. :)

Reply:
Yes.